service-base/src/main/java/org/eclipse/hono/service/AbstractProtocolAdapterBase.java

```diff
@@ -1086,6 +1087,53 @@ protected final Message newMessage(
             final JsonObject registrationInfo,
             final Integer timeUntilDisconnect) {
 
+        return newMessage(target, publishAddress, contentType, payload, tenant, registrationInfo, timeUntilDisconnect,
+                null);
+    }
+
+    /**
+     * Creates a new AMQP 1.0 message.
+     * <p>
+     * Subclasses are encouraged to use this method for creating {@code Message} instances to be sent downstream in
+     * order to have required Hono specific properties being set on the message automatically.
+     * <p>
+     * This method creates a new {@code Message} and sets
+     * <ul>
+     * <li>its <em>content-type</em> to the given value,</li>
+     * <li>its <em>creation-time</em> to the current system time and</li>
+     * <li>its payload as an AMQP <em>Data</em> section</li>
+     * </ul>
+     * The message is then passed to {@link #addProperties(Message, ResourceIdentifier, String, TenantObject, JsonObject, Integer)}.
```

Comment:
it is actually passed to the variant accepting the TTL as well, isn't it?

Reply:
I have updated.